pkg/state/state_skel.go

```diff
@@ -266,12 +266,48 @@ func (s *stateSkel) handleStateObjectsDeletion(ctx context.Context) (SyncState,
 	return SyncStateIgnore, nil
 }
 
-func (s *stateSkel) deleteStateRelatedObjects(ctx context.Context) (bool, error) {
+// remove stale object of the state, returns boolean which indicates if removal is in progress and
+// an error if failed to remove an object
+func (s *stateSkel) handleStaleStateObjects(ctx context.Context,
+	currentObjs []*unstructured.Unstructured) (bool, error) {
+	reqLogger := log.FromContext(ctx)
+	reqLogger.V(consts.LogLevelInfo).Info(
+		"check state for stale objects", "State:", s.name)
+	objsToKeep := stateObjects{}
+	for _, o := range currentObjs {
+		gvkMap := objsToKeep[o.GroupVersionKind()]
```

Comment:
can you make "add" operation a member func of stateObjects ?

Reply:
I added helper function for the stateObjects map